### PR TITLE
Add filter field into dropdowns

### LINKED ISSUE #598: Add filter field into dropdowns
Add a filter field into long dropdowns.

Example (but only add the field, no "please select" text, etc):
image

For now, add to:

1. list of versions
image

2. period selector in all places
image

### PR DESCRIPTION
**Pull request by xbpcb**
Fix #598

### CODE REVIEW

File: src/views/components/periods-dropdown.html [hunk 1]
```diff
@@ -1,36 +1,46 @@
...
 			{{/if}}
 
-			<button
-				type="button"
-				class="btn dropdown-toggle"
-				data-toggle="dropdown"
-				aria-haspopup="true"
-				aria-expanded="false">
-				<span>{{periodToDisplay}}</span>
+			<div class="dropdown_head_selected {{#if dropdownOpened}}opened{{/if}}" on-click="@this.toggleDropdown()">
```
Comment (MartinKolarik) [inline src/views/components/periods-dropdown.html:8]: Why are we replacing a button with a div here? That seems wrong and unnecessary.

File: src/views/components/version-dropdown.html [hunk 2]
```diff
@@ -1,22 +1,84 @@
-<div class="version-dropdown btn-group ">
+<div class="version-dropdown">
 	{{#if versions.length > 1}}
-		<div class="version-dropdown-selected-version" data-toggle="dropdown">
+		<div class="version-dropdown_selected" on-click="@this.toggleDropdown()">
 			<a>{{version}}</a>
 			<img width="20" height="20" src="{{@shared.assetsHost}}/img/icons/chevron-down.svg" loading="lazy"/>
 		</div>
 
-		<ul class="dropdown-menu dropdown-menu-right">
```
Comment (MartinKolarik) [inline src/views/components/version-dropdown.html:8]: Same here. Why replace a good semantical list with divs?

File: src/views/components/version-dropdown.html [hunk 3]
```diff
@@ -1,22 +1,84 @@
...
+		data () {
+			return {
+				filterPlaceholder: 'Find version..',
+				noVersionsMsg: 'No versions found..',
+				dropdownOpened: false,
+			};
+		},
+		oncomplete () {
+			if (!Ractive.isServer) {
+				// close dropdown and clear filter on click outside the dropdown el
```
Comment (MartinKolarik) [inline src/views/components/version-dropdown.html:62]: This seemed to be handled fine before?

**Review by MartinKolarik — COMMENTED** [review]
Let's add it also here: https://dl.dropboxusercontent.com/scl/fi/g5tcbnxhr2h0dny36bpfx/2023-11-10_15-32-59.png?rlkey=sn1k71eh8q8k0keik6l1c92i0&dl=0

File: src/views/components/periods-dropdown.html (hunk 1 above)
Comment (xbpcb) [reply src/views/components/periods-dropdown.html]: Fixed

File: src/views/components/version-dropdown.html (hunk 2 above)
Comment (xbpcb) [reply src/views/components/version-dropdown.html:8]: Fixed

**Comment by MartinKolarik** [comment]
I made some improvements here, but still please add also to: https://dl.dropboxusercontent.com/scl/fi/g5tcbnxhr2h0dny36bpfx/2023-11-10_15-32-59.png?rlkey=sn1k71eh8q8k0keik6l1c92i0&dl=0 (look at the periods dropdown to make sure the keyboard nav handles separators the same way)

Otherwise it looks good.

**Pull request merged**